src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,388 @@
+                    }
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        private static void ParseEntry(XmlNodeList xn, StackTraceInfo stInfo)
+        {
+            try
+            {
+                XmlNode bestPointSoFar = null;
+                long ilOffset = Convert.ToInt64(stInfo.Offset, 16);
+                foreach (XmlNode node in xn)
+                {
+                    // If the attribute is not 'startLine', but 'hidden', select the best value so far
+                    if (Convert.ToInt64(node.Attributes["offset"].Value, 16) > ilOffset)
+                    {
+                        break;
+                    }
+                    if (node.Attributes["startLine"] != null)
+                    {
+                        bestPointSoFar = node;
+                    }
+                }
+                if (bestPointSoFar != null)
+                {
+                    stInfo.StartLine = bestPointSoFar.Attributes["startLine"].Value;
+                    stInfo.EndLine = bestPointSoFar.Attributes["endLine"].Value;
+                    stInfo.Document = bestPointSoFar.Attributes["document"].Value;
+                }
+            }
+            catch (ArgumentException e)
+            {
+                Console.Error.WriteLine(e);
+            }
+        }
+
+        public static Command SymbolicateCommand() =>
+            new Command(
+                name: "symbolicate", description: "Get the line number from the Method Token and IL Offset in a stacktrace")
+            {
+                // Handler
+                HandlerDescriptor.FromDelegate((SymbolicateDelegate)Symbolicate).GetCommandHandler(),
+                // Arguments and Options
+                InputArgument(),
+                SearchDirectoryOption(),
+                OutputOption()
+            };
+
+        public static Argument<FileInfo> InputArgument() =>
+            new Argument<FileInfo>(name: "input-path")
+            {
+                Description = "Path to the stacktrace text file",
+                Arity = ArgumentArity.ExactlyOne
+            }.ExistingOnly();
+
+        public static Option<DirectoryInfo[]> SearchDirectoryOption() =>
+            new Option<DirectoryInfo[]>(new[] { "-d", "--search-dir" }, "Path of multiple directories with assembly and pdb")
+            {
+                Argument = new Argument<DirectoryInfo[]>(name: "directory1 directory2 ...", getDefaultValue: () => new DirectoryInfo(Directory.GetCurrentDirectory()).GetDirectories())
+                {
+                    Arity = ArgumentArity.ZeroOrMore
+                }.ExistingOnly()
+            };
+
+        public static Option<string> OutputOption() =>
+            new Option<string>(new[] { "-o", "--output" }, "Output directly to a file")
+            {
+                Argument = new Argument<string>(name: "output-path")
```

Comment:
FYI, in the latest versions of System.CommandLine, this property is not accessible.